modules/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal

```diff
@@ -0,0 +1,291 @@
+@Return {value:"return base type from MediaType struct"}
+public function <MediaType mediaType> toString () (string) {
+    return mediaType.primaryType + "/" + mediaType.subType;
+}
+
+@Description {value:"Convert the media type to a string suitable for use as the value of a corresponding HTTP header."}
+@Param {value:"mediaType: MediaType struct"}
+@Return {value:"return the Content-Type with parameters as a string"}
+public function <MediaType mediaType> toStringWithParameters () (string) {
+    string contentType = mediaType.toString() + "; ";
+    map parameters = mediaType.parameters;
+    string[] arrKeys = mediaType.parameters.keys();
+    int size = lengthof arrKeys;
+    int index = 0;
+    while (index < size) {
+        var value, _ = (string)parameters[arrKeys[index]];
+        if (index == size - 1) {
+            contentType = contentType + arrKeys[index] + "=" + value;
+            break;
+        } else {
+            contentType = contentType + arrKeys[index] + "=" + value + ";";
+            index = index + 1;
+        }
+    }
+    return contentType;
+}
+
+@Description {value:"Represent MIME specific base64 encoder. This follows RFC 2045 for encoding operation."}
+public struct MimeBase64Encoder {
+}
+
+@Description {value:"Represent MIME specific base64 decoder. This follows RFC 2045 for decoding operation."}
+public struct MimeBase64Decoder {
+}
+
+@Description {value:"Represent quoated-printable encoder."}
+public struct QuotedPrintableEncoder {
+}
+
+@Description {value:"Represent quoated-printable decoder."}
+public struct QuotedPrintableDecoder {
+}
+
+@Description {value:"Encode byte array using MIME Base64 encoding scheme."}
+@Param {value:"encoder: Represent MIME specific base64 encoder"}
+@Param {value:"content: the byte array to encode"}
+@Return {value:"return resulting encoded bytes"}
+public native function <MimeBase64Encoder encoder> encode (blob content) (blob);
+
+@Description {value:"Encode a given string using MIME Base64 encoding scheme. First the given string will be
+converted to a byte array with the given charset encoding. If the charset given is null default 'UTF-8' will be used.
+ Then that byte array will be encoded using MIME Base64 encoding scheme and a new string will be constructed with the
+  given charset."}
+@Param {value:"encoder: Represent MIME specific base64 encoder"}
+@Param {value:"content: string to encode"}
+@Param {value:"charset: charset used in the given string and the resulting string"}
+@Return {value:"return resulting encoded string"}
+public native function <MimeBase64Encoder encoder> encodeString (string content, string charset) (string);
+
+@Description {value:"Decode byte array using MIME Base64 encoding scheme."}
+@Param {value:"encoder: Represent MIME specific base64 decoder"}
+@Param {value:"content: the byte array to decode"}
+@Return {value:"return resulting decoded bytes"}
+public native function <MimeBase64Decoder decoder> decode (blob content) (blob);
```

Comment:
For base 64 encode/decode, can't we make use of the existing functions provided in `ballerina.util` package?

Reply:
This is mime specific base64 encoder/decoder. There's a difference between the two so cannot use the existing functions.